framework/src/utils/MooseObjectParameterName.C

```diff
@@ -0,0 +1,126 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "MooseObjectParameterName.h"
+
+// STL includes
+#include <iostream>
+
+MooseObjectParameterName::MooseObjectParameterName(const MooseObjectName & obj_name, std::string param) :
+    MooseObjectName(obj_name),
```

Comment:
OK, can you add additional comments in the headers of these two classes explaining the two formats? I see some comments in the body for each subroutine but they don't really explaining the full definition of these two formats